Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyFulfillmentAPI.Codeunit.al

```diff
@@ -0,0 +1,53 @@
+namespace Microsoft.Integration.Shopify;
+
+/// <summary>
+/// Codeunit Shpfy Fulfillment API (ID 30315).
+/// </summary>
+codeunit 30315 "Shpfy Fulfillment API"
+{
+    var
+        ShpfyJsonHelper: Codeunit "Shpfy Json Helper";
+        ShpfyCommunicationMgt: Codeunit "Shpfy Communication Mgt.";
+
+    /// <summary>
+    /// Fulfills shopify orders for each fulfillment id parsed from a completed draft order.
+    /// </summary>
+    /// <param name="JResponse">Json response from a completed draft order</param>
+    /// <param name="ShopCode">Shopify shop code to be used.</param>
+    internal procedure FulfillShopifyOrder(JResponse: JsonToken; ShopCode: Code[20])
+    var
+        FulfillmentOrderList: List of [Text];
+        FulfillmentOrderId: Text;
+        ResponseJsonToken: JsonToken;
+        GraphQLType: Enum "Shpfy GraphQL Type";
+        Parameters: Dictionary of [Text, Text];
+    begin
+        ShpfyCommunicationMgt.SetShop(ShopCode);
+        FulfillmentOrderList := ParseFulfillmentOrders(JResponse);
+        GraphQLType := "Shpfy GraphQL Type"::FulfillOrder;
+
+        foreach FulfillmentOrderId in FulfillmentOrderList do begin
+            Parameters.Add('FulfillmentOrderId', FulfillmentOrderId);
+            ResponseJsonToken := ShpfyCommunicationMgt.ExecuteGraphQL(GraphQLType, Parameters);
+            Clear(Parameters);
+        end;
+    end;
+
+    local procedure ParseFulfillmentOrders(ResponseJsonToken: JsonToken) FulfillmentOrderList: List of [Text]
```

Comment:
This one should be in the orchestrator as it specifically only parses fulfilment orders out of a draftOrderComplete